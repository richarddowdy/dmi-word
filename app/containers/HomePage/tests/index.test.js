/**
 *
 * Tests for Home
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import configureStore from '../../../configureStore';

// import 'jest-dom/extend-expect'; // add some helpful assertions

import { HomePage } from '../index';
import { DEFAULT_LOCALE } from '../../../i18n';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    const onPageLoad = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <HomePage dispatch={dispatch} onPageLoad={onPageLoad} />
        </IntlProvider>
      </Provider>,
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should call onPageLoad when the page Loads', () => {
    const spy = jest.spyOn(global.console, 'error');
    const dispatch = jest.fn();
    const onPageLoad = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <HomePage dispatch={dispatch} onPageLoad={onPageLoad} />
        </IntlProvider>
      </Provider>,
    );
    expect(onPageLoad).toHaveBeenCalled();
    expect(spy).not.toHaveBeenCalled();
  });

  /**
   * Unskip this test to use it
   *
   * @see {@link https://jestjs.io/docs/en/api#testskipname-fn}
   */
  it('Should render and match the snapshot', () => {
    const onPageLoad = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <HomePage onPageLoad={onPageLoad} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
