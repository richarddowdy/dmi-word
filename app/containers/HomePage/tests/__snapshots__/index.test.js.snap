// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> Should render and match the snapshot 1`] = `
<div>
  <header
    class="Header-sc-1lzyagy-0 hXcZuE"
  >
    <h1
      class="H1-tncdg6-0 gRRKTv"
    >
      The Lord of the 
      <small>
        st
      </small>
      <span
        style="font-size: 90px;"
      >
        R
      </span>
      ings
    </h1>
  </header>
  <section
    class="Section-sc-1eulrsm-0 fuRJUh"
  >
    <div>
      <i
        class="fas fa-spinner fa-pulse fa-3x"
      />
    </div>
  </section>
</div>
`;
